quote-server/server.js

```diff
@@ -1,32 +1,50 @@
-// server.js
-// This is where your node app starts
-
-//load the 'express' module which makes writing webservers easy
 import express from "express";
-//load the quotes JSON
 import quotes from "./quotes.json" assert { type: "json" };
+import cors from "cors";
 
 const app = express();
-// Now register handlers for some routes:
-//   /                  - Return some helpful welcome info (text)
-//   /quotes            - Should return all quotes (json)
-//   /quotes/random     - Should return ONE quote (json)
+
+app.use(cors());
+
 app.get("/", (request, response) => {
   response.send("Neill's Quote Server!  Ask me for /quotes/random, or /quotes");
 });
 
-//START OF YOUR CODE...
+app.get("/quotes", (req, res) => {
+  res.json(quotes);
+});
+
+app.get("/quotes/random", (req, res) => {
+  const randomQuote = pickFromArray(quotes);
+  res.json(randomQuote);
+});
+// example http://localhost:3001/quotes/search?term=day
 
-//...END OF YOUR CODE
+app.get("/quotes/search", (request, response) => {
+  const searchTerm = request.query.term;
+  console.log("Search Term:", searchTerm);
 
-//You can use this function to pick one element at random from a given array
-//example: pickFromArray([1,2,3,4]), or
-//example: pickFromArray(myContactsArray)
-//
+  const matchingQuotes = quotes.filter((quote) => {
+    const quoteText = quote.quote.toLowerCase();
+    const authorName = quote.author.toLowerCase();
+    const searchTermLower = searchTerm.toLowerCase();
```

Comment:
How many times will `searchTerm.toLowerCase()` get called when you do a search? Will the returned value be different between the different calls? Can you think of a way to call this function fewer times?